### DISCUSSION TITLE
Carbon footprint calculation for web/UI application

### DISCUSSION THREAD

**Discussion by navveenb** [Green-Software-Foundation/sci#105] [General]
I wanted to check if the SCI Specification would cover how to calculate carbon footprint of web application - specifically how the UI application is being rendered. For instance, it's can be based on bytes transferred , size of the web page, network latency/hops etc (like served from edge). A web page having lot of high resolution images, videos, blocking scripts etc.. being rendered leads to a higher carbon footprint. 

We have additional constraints on finding out energy consumption of various devices/mobiles., the region electricity factors needs to be considered, but is there a way we can build some bechmark or guideline.  This can help to estimate and reduce to carbon footprint of the web app..

User interface and presentation layer is an important element of any application, it would be good to have this covered as part of the specification or may be probably in FAQ.

**Comment by atg-abhishek** [comment]
@navveenb building on #100 , I would say that this can go as a webpage on the wiki with a separate page for measuring the carbon impacts of web apps. @SaraEmilyBergman also has a ton of great resources to bootstrap that I would imagine :)

**Comment by SaraEmilyBergman** [comment]
Very interesting question and a hard topic indeed!
I have written two blog posts on this topic, there are some links in them:
- https://devblogs.microsoft.com/sustainable-software/how-to-measure-the-energy-consumption-of-your-apps/
- https://devblogs.microsoft.com/sustainable-software/how-to-measure-the-power-consumption-of-your-frontend-application/

Specifically these two papers are interesting: https://ts.data61.csiro.au/publications/papers/Carroll:phd.pdf and https://dl.acm.org/doi/abs/10.1145/2602044.2602054

@pierlag also has a great blog post on the topic: https://devblogs.microsoft.com/sustainable-software/measuring-the-carbon-impact-of-web-browsing/ with great links in it :)

**Comment by jawache** [comment]
I'm glad you asked this question @navveenb, I worry too that we are too leaning towards cloud and need to make sure we cover client apps. I think the SCI can cover both, the latest PR https://github.com/Green-Software-Foundation/software_carbon_intensity/pull/96 I think specifically addresses making sure the SCI can be used for both client and cloud apps.

I think we need to pick several examples and create a more detailed article for how we would calculate the SCI for that example. A web app is an interesting one since it has both a client and a cloud component, so an article describing how you would calculate the SCI of a web app would cover a lot of use cases.
